### PR TITLE
add eslint-scope

### PR DESCRIPTION
**Pull request by mysticatea**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not provide its own types, and you can not add them.
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/eslint-scope/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
+// Type definitions for eslint-scope 3.7
+// Project: http://github.com/eslint/eslint-scope
+// Definitions by: Toru Nagashima <https://github.com/mysticatea>
```
Comment (DanielRosenwasser) [inline types/eslint-scope/index.d.ts]: Can you switch the bracketed form to `https://github.com/mysticatea`?

File: types/eslint-scope/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import * as eslint from "eslint";
+import * as estree from "estree";
+
+export const version: string;
+
+export class ScopeManager implements eslint.Scope.ScopeManager {
+    scopes: Scope[];
+    globalScope: Scope;
+    acquire(node: {}, inner?: boolean): Scope | null;
+    getDeclaredVariables(node: {}): Variable[];
```
Comment (DanielRosenwasser) [inline types/eslint-scope/index.d.ts]: Instead of `{}`, why not use an estree Node? (here and in other locations)

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

File: types/eslint-scope/index.d.ts (hunk 2 above)
Comment (mysticatea) [reply types/eslint-scope/index.d.ts]: ESLint AST could not assign to `@types/estree` AST types, so I'd like to keep relax here.
(though I have ESLint AST definitions in my project local [vue-eslint-parser/src/ast/nodes.ts](https://github.com/mysticatea/vue-eslint-parser/blob/6111437a4b470fb63227099a69749a7ef2f022a4/src/ast/nodes.ts), I want to find why it cannot assign to estree types. But the type definition is recursive, so `tsc` doesn't show meaningful error :cry:)

Comment (mysticatea) [reply types/eslint-scope/index.d.ts]: `eslint-scope` uses those parameters as only the key of `Map`, so I think this relaxing does not trigger runtime errors.

**Review by DanielRosenwasser — APPROVED** [review]

**Pull request merged**